test/integration/framework/shoot_operations.go

```diff
@@ -0,0 +1,270 @@
+
+package framework
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/gardener/gardener/pkg/apis/garden/v1beta1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/tools/cache"
+
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+// NewShootGardenerTest creates a new shootGardenerTest object, given an already created shoot (created after parsing a shoot YAML)
+func NewShootGardenerTest(kubeconfig string, shoot *v1beta1.Shoot, logger *logrus.Logger) (*ShootGardenerTest, error) {
+	if len(kubeconfig) == 0 {
+		return nil, fmt.Errorf("please specify the kubeconfig path correctly")
+	}
+
+	gardenerTest, err := newGardenerTest(kubeconfig)
+	if err != nil {
+		return nil, err
+	}
+
+	// create shoot Informers and Listers
+	shootInformer := gardenerTest.K8sGardenInformers.Garden().V1beta1().Shoots()
+	shootLister := shootInformer.Lister()
+
+	seedInformer := gardenerTest.K8sGardenInformers.Garden().V1beta1().Seeds()
+	seedLister := seedInformer.Lister()
+
+	projectInformer := gardenerTest.K8sGardenInformers.Garden().V1beta1().Projects()
+	projectLister := projectInformer.Lister()
+
+	cloudprofileInformer := gardenerTest.K8sGardenInformers.Garden().V1beta1().CloudProfiles()
+	cloudprofileLister := cloudprofileInformer.Lister()
+
+	secretBindingInformer := gardenerTest.K8sGardenInformers.Garden().V1beta1().SecretBindings()
+	secretBindingLister := secretBindingInformer.Lister()
+
+	secretsInformer := gardenerTest.KubeInformerFactory.Core().V1().Secrets()
+	secretsLister := secretsInformer.Lister()
+
+	// Start generating the GardenTestOperation
+	ctx := context.Background()
+	gardenerTest.K8sGardenInformers.Start(ctx.Done())
+
+	if !cache.WaitForCacheSync(ctx.Done(),
+		shootInformer.Informer().HasSynced,
+		seedInformer.Informer().HasSynced,
+		projectInformer.Informer().HasSynced,
+		cloudprofileInformer.Informer().HasSynced,
+		secretBindingInformer.Informer().HasSynced) {
+		panic("Timed out waiting for Garden caches to sync")
+	}
+
+	gardenerTest.KubeInformerFactory.Start(ctx.Done())
+	if !cache.WaitForCacheSync(ctx.Done(), secretsInformer.Informer().HasSynced) {
+		panic("Timed out waiting for Kube caches to sync")
+	}
+
+	return &ShootGardenerTest{
+		GardenerTest: *gardenerTest,
+
+		SecretsInformer: secretsInformer,
+		SecretsLister:   secretsLister,
+
+		SecretBindingInformer: secretBindingInformer,
+		SecretBindingLister:   secretBindingLister,
+
+		CloudProfileInformer: cloudprofileInformer,
+		CloudProfileLister:   cloudprofileLister,
+
+		ProjectInfomer: projectInformer,
+		ProjectLister:  projectLister,
+
+		ShootInformer: shootInformer,
+		ShootLister:   shootLister,
+
+		SeedInformer: seedInformer,
+		SeedLister:   seedLister,
+
+		Shoot:  shoot,
+		Logger: logger,
+	}, nil
+}
+
+// ShootExists returns true if a shoot with the same name exists
+func (s *ShootGardenerTest) ShootExists(ctx context.Context) (bool, *v1beta1.Shoot, error) {
```

Comment:
Why does a `ShootExists` also return the shoot?

Reply:
Because shoot exists gets the actual shoot object, because the one injected is just an ObjectKey NamespacedName shoot, which serves the purpose of finding the shoot only.

Reply:
Yes but then the signature still should be `(bool, error)`.